setup.py

```diff
@@ -0,0 +1,113 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+
+import re
+import shutil
+from pathlib import Path
+
+from setuptools import find_packages, setup
+from setuptools.command.install import install
+
+# Remove stale QEfficient.egg-info directory to avoid issues
+stale_egg_info = Path(__file__).parent / "QEfficient.egg-info"
+if stale_egg_info.exists():
+    print(f"Removing stale {stale_egg_info}")
+    shutil.rmtree(stale_egg_info)
+
+_deps = [
+    "filelock",
+    "numpy>=1.23.0",
+    "packaging",
+    "protobuf==3.20.2",
+    "pyyaml",
+    "regex!=2019.12.17",
+    "requests",
+    "tqdm",
+    "transformers==4.44.2",
+    "torch==2.0.0+cpu",
+    "peft==0.12.0",
+    "datasets==2.7.0",
+    "fsspec==2023.6.0",
+    "multidict==6.0.4",
+    "urllib3<2",
+    "sentencepiece==0.1.98",
+    "onnx==1.16.0",
+    "onnxruntime==1.16.3",
+    "onnxscript==0.1.0.dev20240327",
+    "sympy",
+]
+
+deps = {b: a for a, b in (re.findall(r"^(([^!=<>~ ]+)(?:[!=<>~ ].*)?$)", x)[0] for x in _deps)}
+
+
+def deps_list(*pkgs):
+    return [deps[pkg] for pkg in pkgs]
+
+
+extras = {}
+extras["test"] = ["pytest", "pytest-mock"]
+extras["docs"] = ["Sphinx==7.1.2", "sphinx-rtd-theme==2.0.0", "myst-parser==3.0.1"]
+extras["quality"] = ["black", "ruff", "hf_doc_builder@git+https://github.com/huggingface/doc-builder.git"]
```

Comment:
We can have everything as a single dict definition?

Reply:
Yes, we can, `nit`.
Will test and push once.